js/packages/eyes-sdk-core/lib/sdk/CheckSettingsUtils.js

```diff
@@ -196,60 +196,74 @@ function toMatchSettings({checkSettings = {}, configuration}) {
   }
 }
 
-async function toScreenshotCheckSettings({checkSettings, context, screenshot}) {
-  const screenshotCheckSettings = {
-    ...checkSettings,
-    ignoreRegions: await referencesToRegions(checkSettings.ignoreRegions),
-    floatingRegions: await referencesToRegions(checkSettings.floatingRegions),
-    strictRegions: await referencesToRegions(checkSettings.strictRegions),
-    layoutRegions: await referencesToRegions(checkSettings.layoutRegions),
-    contentRegions: await referencesToRegions(checkSettings.contentRegions),
-    accessibilityRegions: await referencesToRegions(checkSettings.accessibilityRegions),
+function toBaseCheckSettings(settings) {
+  const regionTypes = ['ignore', 'layout', 'strict', 'content', 'floating', 'accessibility']
+  const regions = regionTypes.flatMap(regionType => {
+    return (settings[`${regionType}Regions`] ? settings[`${regionType}Regions`] : []).reduce((regions, reference) => {
+      const {region} = utils.types.has(reference, 'region') ? reference : {region: reference}
+      return !isRegion(region) ? regions.concat(region) : regions
+    }, [])
+  })
+
+  function getBaseCheckSettings(regionsByCoordinates) {
+    const transformedSettings = {...settings}
+    for (const regionType of regionTypes) {
+      const enrichedRegions = []
+      transformedSettings[`${regionType}Regions`] = transformedSettings[`${regionType}Regions`]
+        ? transformedSettings[`${regionType}Regions`]
+        : []
+      for (const region of transformedSettings[`${regionType}Regions`]) {
```

Comment:
if the point of this is just to have something to use in `for`, then better do this 
```suggestion
      for (const region of transformedSettings[`${regionType}Regions`] || []) {
```